File: rating 800/Maths/SpyDetected.java
import java.util.*;
import java.io.*;

public class SpyDetected{
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;

    static String next() throws IOException{
        while(st == null || !st.hasMoreTokens()){
            st = new StringTokenizer(br.readLine());
        }
        return st.nextToken();
    }

    static int nextInt() throws IOException{
        return Integer.parseInt(next());
    }

    static int solve() throws IOException{
        int n = nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++){
            arr[i] = nextInt();
        }
        int majority = 0;

        if(arr[0] == arr[1] || arr[0] == arr[2]){
            majority = arr[0];
        }
        else majority = arr[1];

        for(int i = 0; i < n; i++){
            if(arr[i] != majority){
                return i + 1;
            }
        }
        return -1;
    }

    public static void main(String[] args) throws IOException{
        int t = nextInt();
        while(t-- > 0){
            System.out.println(solve());
        }
    }

}

File: Kotlin-Heros/SpyDetected.kt
import java.io.*;
import java.util.*;

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val t = br.readLine().toInt()

    repeat(t){
        val n = br.readLine().toInt()
        val parts = StringTokenizer(br.readLine())
        val array = IntArray(n){
            parts.nextToken().toInt()
        }

        if(array[0] == array[1]){
            for(i in 2 until n){
                if(array[i] != array[0]){
                    println(i + 1)
                    break
                }
            }
        }
        else{
            if(array[0] == array[2]){
                println(2);
            }
            else println(1);
        }
    }
}